### ISSUE TITLE
Primary carbamates cleaning

### ISSUE THREAD

**Issue by Adafede** [nuzillard/KnapsackSearch#1]
Hi @nuzillard!

First of all, thank you very much for using LOTUS!

After reading your [paper](https://doi.org/10.3390/analytica2030006), and having some other feedback concerning our depiction of, more particularly, amide bonds, I decided to integrate your script in our processing pipeline ([here](https://github.com/lotusnprod/lotus-processor/blob/1c53487cc67440eab28800a116a35af55b811929/src/2_curating/2_editing/structure/3_processingAndEnriching/chemosanitizer_functions.py#L155)). The next LOTUS frozen release should so have this corrected.

The reason for this issue is actually that I could not manage to run the primary carbamates tautomers cleaning, as it always returns an empty tuple. I had no time to investigate it further for now and will post any update on this here, but wanted to share it with you in case!

Best,

**Comment by nuzillard** [comment]
Hi @Adafede!

Yes, LOTUS is an utterly useful information source for structural natural product chemistry.

The file in [primcarbam.zip](https://github.com/nuzillard/KnapsackSearch/files/8516367/primcarbam.zip) attempts to reproduce the issue using methyl carbamate as example but the result is the expected one.

**Comment by Adafede** [comment]
Thank you very much for your example!

It actually looks like the SMILES is already the same at the beginning (from print())?

Here  attached some examples that failed on my side:

[primcarbam_2.py.zip](https://github.com/nuzillard/KnapsackSearch/files/8518420/primcarbam_2.py.zip)

**Comment by nuzillard** [comment]
The output of my demo script is:
Initially: COC(=N)O
1 primary carbamate
COC(N)=O
Finally: COC(N)=O
The initially "badly placed" double bond (C=N) was moved finally to the "right place" (C=O).

The examples you provided and that failed contain an *aromatic* primary carbamate (derived from a phenol).
The reaction SMARTS was written for aliphatic primary carbamates only.
A quick fix is to change the reaction SMARTS to:
'[#6:1][O:2][CD3:3]([OH1:4])=[NH:5]>>[#6:1][O:2][CD3:3](=[OH0D1:4])[NH2:5]'
The code in tautomer.py probably requires some cleaning.
The issue will be closed once cleaning done.

See also:
https://doi.org/10.1186/s13321-020-00456-1
https://doi.org/10.1186/s13321-018-0293-8

**Comment by Adafede** [comment]
Right, I misread the output and only focused on the two identical last lines...my bad!

Indeed, the aromatic was the problem!

Replacing by `rxn11 = AllChem.ReactionFromSmarts('[#6:1][O:2][CD3:3]([OH1:4])=[NH:5]>>[#6:1][O:2][CD3:3](=[OH0D1:4])[NH2:5]')` seems to work! 

Many thanks! 😊

**Comment by Adafede** [comment]
(sorry for the spam)

Just had an interesting 

`N=C(OP(O)(O)=O)O`

so I more simply went for `'[O:2][CD3:3]([OH1:4])=[NH:5]>>[O:2][CD3:3](=[OH0D1:4])[NH2:5]'`

**Comment by nuzillard** [comment]
Found that one as well: https://github.com/WahlOya/Tautobase

**Comment by nuzillard** [comment]
[https://pubs.acs.org/doi/pdf/10.1021/acs.jcim.9b01080](https://pubs.acs.org/doi/pdf/10.1021/acs.jcim.9b01080)
for a long list of tautomerisation rules. Thanks to Chicago Ji <chicago.ji@gmail.com>
and her/his recent post on rdkit-discuss.
Can be tested with [https://cactus.nci.nih.gov/tautomerizer](https://cactus.nci.nih.gov/tautomerizer).

**Comment by Adafede** [comment]
Very interesting resource indeed!

Thank you for sharing it!

**Comment by nuzillard** [comment]
A recent dump of LOTUS (May 18, 2022) as a .smi file through the web interface contains 276,519 lines among which 208 have a SMILES containg the dot (.) character.
At line 707:
COc1nc(C)c(=O)n(O)c1Cc1c[nH]c2c(CC=C(C)C)cccc12.COc1nc(C)c(=O)n(O)c1Cc1c[nH]c2c(CC=C(C)C)cccc12.COc1nc(C)c(=O)n(O)c1Cc1c[nH]c2c(CC=C(C)C)cccc12 	LTS0086317
there is no tautomer substitution to carry out and this is fine, even though the same molecule appears in triplicate.
At line 44603:
CC(C)(CO)[C@@H](O)C(O)=NCCC(=O)O.CC(C)(CO)[C@@H](O)C(O)=NCCC(=O)O 	LTS0013066
there are two secondary iminols and this wreaks havoc in tautomer substitution.

**Comment by Adafede** [comment]
Thank you!

Sadly, the web interface is not up-to-date (yet!), but we will push an update soon(ish) that should include correction of these aspects (and others). 

If this helps, I can share a pre-release version with you so you can check!

A the moment, most up-to-date versions are on https://zenodo.org/communities/the-lotus-initiative/?page=1&size=20, even it the one with the corrections I am mentioning is not there yet!
